common/mariadb/templates/initdb/_init.sql.tpl

```diff
@@ -1,3 +1,25 @@
-CREATE DATABASE IF NOT EXISTS {{.Values.name}} CHARACTER SET utf8 COLLATE utf8_general_ci;
-GRANT ALL PRIVILEGES ON {{.Values.name}}.* TO {{.Values.global.dbUser}}@localhost IDENTIFIED BY '{{include "db_password" .}}';
-GRANT ALL PRIVILEGES ON {{.Values.name}}.* TO {{.Values.global.dbUser}}@'%' IDENTIFIED BY '{{include "db_password" .}}';
+{{- if not .Values.databases }}
+CREATE DATABASE IF NOT EXISTS {{ .Values.name }};
+{{- else }}
+    {{- range .Values.databases }}
+CREATE DATABASE IF NOT EXISTS {{ . }};
+    {{- end }}
+{{- end }}
+
+{{- if and .Values.global.dbUser .Values.global.dbPassword }}
+CREATE USER IF NOT EXISTS {{ .Values.global.dbUser }};
+GRANT ALL PRIVILEGES ON {{ .Values.name }}.* TO {{ .Values.global.dbUser }} IDENTIFIED BY {{ include "db_password" . }};
```

Comment:
Do I understand that correct that in case of multiple Values.databases we are not adding the permissions?

Reply:
Yes, `.Values.databases` only creates the databases.
The users and their grants are created & maintained by `.Values.users` and `.Values.users[*].grants` respectively.

We can add some defaults, which avoids repetitive defaults, the issue though here is, that such a logic internal to this chart would be needed to replicated to anyone trying to evaluate the settings (i.e. the proxysql side-car,  PR #3554).

Certainly doable, but I think having to be explicit once in the parent chart is not a lot of effort. I would take care of it.